packages/langium/src/test/langium-test.ts

```diff
@@ -769,3 +768,103 @@ export function expectSemanticToken(tokensWithRanges: DecodedSemanticTokensWithR
     });
     expectedFunction(result.length, 1, `Expected one token with the specified options but found ${result.length}`);
 }
+
+export interface QuickFixResult<T extends AstNode = AstNode> extends AsyncDisposable {
+    /** the document containing the AST */
+    document: LangiumDocument<T>;
+    /** all diagnostics of the validation */
+    diagnosticsAll: Diagnostic[];
+    /** the relevant Diagnostic with the given diagnosticCode, it is expected that the given input has exactly one such diagnostic */
+    diagnosticRelevant: Diagnostic;
+    /** the CodeAction to fix the found relevant problem, it is possible, that there is no such code action */
+    action?: CodeAction;
+}
+
+/**
+ * This is a helper function to easily test quick-fixes for validation problems.
+ * @param services the Langium services for the language with quick fixes
+ * @returns A function to easily test a single quick-fix on the given invalid 'input'.
+ * This function expects, that 'input' contains exactly one validation problem with the given 'diagnosticCode'.
+ * If 'outputAfterFix' is specified, this functions checks, that the diagnostic comes with a single quick-fix for this validation problem.
+ * After applying this quick-fix, 'input' is transformed to 'outputAfterFix'.
+ */
+export function testQuickFix<T extends AstNode = AstNode>(services: LangiumServices): (input: string, diagnosticCode: string, outputAfterFix: string | undefined, options?: ParseHelperOptions) => Promise<QuickFixResult<T>> {
```

Comment:
Suggestion: I would keep the `CodeAction` naming convention from the LSP instead of using `QuickFix` (which we don't use anywhere else in the code base).